tests/test_pyathenajdbc.py

```diff
@@ -224,7 +224,7 @@ def test_output_location(self, cursor):
         self.assertIsNone(cursor.output_location)
         cursor.execute('SELECT * from one_row')
         self.assertEqual(cursor.output_location,
-                         '{0}{1}.csv'.format(ENV.s3_staging_dir, cursor.query_id))
+                         '{0}/{1}.csv'.format(ENV.s3_staging_dir, cursor.query_id))
```

Comment:
s3_staging_dir does not end with a slash?
If it ends with a slash, output_location will be `s3://path/to//123456789.csv`.
os.path.join It looks good, but the test is likely to fail on Windows.

Reply:
Hmm. I get 19 test failures per environment on my Mac when I add the '/' to the `s3_staging_dir`.

Reply:
That's why I added '/' to the test itself. I don't have a Windows machine currently available but I can run the tests on a Linux VM too.

Reply:
Here are the tests that fail for me if I add a trailing '/' to the `s3_staging_dir`:

```
tests/test_pyathenajdbc.py::TestPyAthenaJDBC::test_arraysize FAILED
tests/test_pyathenajdbc.py::TestPyAthenaJDBC::test_cancel FAILED
tests/test_pyathenajdbc.py::TestPyAthenaJDBC::test_complex FAILED
tests/test_pyathenajdbc.py::TestPyAthenaJDBC::test_contain_special_character_query FAILED
tests/test_pyathenajdbc.py::TestPyAthenaJDBC::test_contain_special_character_query_with_parameter FAILED
tests/test_pyathenajdbc.py::TestPyAthenaJDBC::test_escape FAILED
tests/test_pyathenajdbc.py::TestPyAthenaJDBC::test_fetchall FAILED
tests/test_pyathenajdbc.py::TestPyAthenaJDBC::test_fetchmany FAILED
tests/test_pyathenajdbc.py::TestPyAthenaJDBC::test_fetchone FAILED
tests/test_pyathenajdbc.py::TestPyAthenaJDBC::test_iterator FAILED
tests/test_pyathenajdbc.py::TestPyAthenaJDBC::test_multiple_connection FAILED
tests/test_pyathenajdbc.py::TestPyAthenaJDBC::test_null FAILED
tests/test_pyathenajdbc.py::TestPyAthenaJDBC::test_null_param FAILED
tests/test_pyathenajdbc.py::TestPyAthenaJDBC::test_unicode FAILED
tests/test_sqlalchemy_athena.py::TestSQLAlchemyAthena::test_basic_query FAILED
tests/test_sqlalchemy_athena.py::TestSQLAlchemyAthena::test_char_length FAILED
tests/test_sqlalchemy_athena.py::TestSQLAlchemyAthena::test_reflect_select FAILED
tests/test_sqlalchemy_athena.py::TestSQLAlchemyAthena::test_unicode FAILED
```

I didn't see an obvious connection between these tests on first glance. I can look a bit deeper later today.

Reply:
The environment variable `AWS_ATHENA_S3_STAGING_DIR` is also used for uploading test data.
https://github.com/laughingman7743/PyAthenaJDBC/blob/master/scripts/test_data/upload_test_data.sh
This value assumes that there is a trailing slash.